### ISSUE TITLE
Pool thread not daemon

### ISSUE THREAD

**Issue by fzakaria** [jruby/jruby#6022]
**tl;dr;** There is a non-daemon thread persisting causing the JVM to not fully shutdown.

I've opened an issue to _concurrent-ruby_ (https://github.com/ruby-concurrency/concurrent-ruby/issues/839) but it may actually be a JRuby issue since the thread in question looks like a JRuby framework owned-thread. I would have guessed (and I believe read) that all threads in JRuby should be daemonized.

I have a minimal reproduction at https://github.com/fzakaria/concurrent-ruby-timer-task-failure of leveraging  _Concurrent::TimerTask_ which causes the failure.

```ruby
require 'minitest/autorun'
require 'concurrent-ruby'

class TimerTaskTest < Minitest::Test

	def test_start_timer_task_shutdown
		task = Concurrent::TimerTask.new do
		end
		task.execute
	end
end
```

Below is a snapshot of a stacktrace I found from doing `jstack` and finding the _only_ non-daemonized thread in the output.

```
"pool-6-thread-1" #61 prio=5 os_prio=31 tid=0x00007fe599823800 nid=0xb103 in Object.wait() [0x0000700013a56000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:460)
        at org.jruby.RubyThread$SleepTask.run(RubyThread.java:1410)
        - locked <0x000000076e124a08> (a com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject)
        at org.jruby.RubyThread.executeBlockingTask(RubyThread.java:1463)
        at org.jruby.RubyThread.wait_timeout(RubyThread.java:1933)
        at com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject.nsWait(SynchronizationLibrary.java:254)
        at com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject$INVOKER$i$0$1$nsWait_DBG.call(SynchronizationLibrary$JRubyLockableObject$INVOKER$i$0$1$nsWait_DBG.gen)
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:796)
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:202)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:153)
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:315)
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:159)
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:127)
        at org.jruby.runtime.InterpretedIRBlockBody.yieldDirect(InterpretedIRBlockBody.java:109)
        at org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:80)
        at org.jruby.runtime.Block.yieldSpecific(Block.java:134)
        at org.jruby.RubyKernel.loop(RubyKernel.java:1316)
        at org.jruby.RubyKernel$INVOKER$s$0$0$loop_DBG.call(RubyKernel$INVOKER$s$0$0$loop_DBG.gen)
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:536)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:77)
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:84)
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:429)
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:360)
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:159)
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:84)
        at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(InterpretedIRMethod.java:182)
        at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(InterpretedIRMethod.java:165)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:163)
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:170)
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:336)
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:159)
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:127)
        at org.jruby.runtime.InterpretedIRBlockBody.yieldDirect(InterpretedIRBlockBody.java:109)
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:114)
        at org.jruby.runtime.Block.yield(Block.java:165)
        at com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject.rubySynchronize(SynchronizationLibrary.java:232)
        - locked <0x000000076e124a08> (a com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject)
        at com.concurrent_ruby.ext.SynchronizationLibrary$JRubyLockableObject$INVOKER$i$0$0$rubySynchronize_DBG.call(SynchronizationLibrary$JRubyLockableObject$INVOKER$i$0$0$rubySynchronize_DBG.gen)
```

**Environment**
  
Provide at least:
```jruby 9.1.17.0 (2.3.3) 2018-04-20 d8b1ff9 OpenJDK 64-Bit Server VM 25.232-b09 on 1.8.0_232-b09 +jit [darwin-x86_64]

Darwin Farid-Zakarias-MacBook-Pro.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64
```

Gem installed is: **concurrent-ruby**

Minimal reproduction example: https://github.com/fzakaria/concurrent-ruby-timer-task-failure

**Expected Behavior**

After MiniTest is finished & the main thread exits, the JVM _should_ shutdown.

**Actual Behavior**

The JVM **does not** shutdown and hangs waiting for a non-daemon thread to exit.

**Comment by fzakaria** [comment]
This was a `concurrent-ruby` issue  -- sorry for the spam.
I've contributed https://github.com/ruby-concurrency/concurrent-ruby/pull/840 to them.

**Comment by headius** [comment]
@fzakaria Thank you! I was about to point out https://github.com/ruby-concurrency/concurrent-ruby/issues/817 but you already found and fixed it. Here's hoping it can get merged and released soon!

**Comment by fzakaria** [comment]
@headius 👍 
I was hoping for a response by now!
The bug is pretty debilitating and basically kills any chance of incorporating it if we use minitest.

Looking through the source the whole registration for exit handlers to shutdown vs. just setting all threads as daemons is perplexing
